### ISSUE TITLE
@stylistic/declaration-colon-space-before and @stylistic/declaration-colon-space-after act on a colon inside a comment

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#92]
Found while fixing #88. Both rules share `declarationColonSpaceChecker`, which scans for the declaration's colon by taking the **first** `:` in the string and breaking out of the loop there. A comment standing between the property and the colon may hold a colon of its own — every URL does — and from then on both rules work on the wrong character.

### What minimal example or steps are needed to reproduce the bug?

```css
a { color/* https://foo.bar/ */:pink; }
```

A URL is only the most common case; a bare `/* a:b */` reproduces it just as well.

### What minimal configuration is needed to reproduce the bug?

```json
{
  "plugins": "@stylistic/stylelint-plugin",
  "rules": {
    "@stylistic/declaration-colon-space-before": "always"
  }
}
```

`@stylistic/declaration-colon-space-after` reproduces it with the same input, and both rules reproduce it with `never` as well.

### What did you expect to happen?

The whitespace around the declaration's own colon is checked and fixed, and the contents of the comment are none of the rule's business.

### What actually happened?

With `always`, the fix edits the text of the comment and leaves the reported violation exactly where it was:

```css
/* declaration-colon-space-before: "always" */
a { color/* https ://foo.bar/ */:pink; }

/* declaration-colon-space-after: "always" */
a { color/* https: //foo.bar/ */:pink; }
```

Note that the warning itself points at the right place — `1:11`, the declaration's colon — because the reported index is derived from the property's length. Only the fix goes to the comment. Running `--fix` twice therefore keeps chewing through the comment, one space per run, while the warning never goes away.

With `never` there is no warning at all, since the colon inside the comment already satisfies the rule:

```css
/* declaration-colon-space-before: "never" — no warning, the real violation is invisible */
a { color/* https://foo.bar/ */ :pink; }

/* declaration-colon-space-after: "never" — likewise */
a { color/* https://foo.bar/ */: pink; }
```

The cause is at [`lib/utils/declarationColonSpaceChecker/index.js:41-42`](https://github.com/stylelint-stylistic/stylelint-stylistic/blob/main/lib/utils/declarationColonSpaceChecker/index.js#L41-L42):

```js
for (let i = 0, l = propPlusColon.length; i < l; i += 1) {
	if (propPlusColon[i] !== `:`) continue
```

### Do you have a proposal to fix the bug?

Mask the comments in `propPlusColon` before scanning — replace the characters of every `/* … */` (and of every `//` comment up to its line break, for SCSS-like syntaxes) with a filler of the same length, so that indices still line up with the original string. The loop then finds the declaration's own colon, and both the reported position and the index handed to the rules' `fix` describe the same character.

Both rules need test cases: a comment with an URL and a comment with a bare colon, under `always` and `never`, checking the warning and the fixed output.
